Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -232,6 +232,26 @@ struct SymbolGraphLoader {
             for (selector, _) in symbol.mixins {
                 if var symbolAvailability = (symbol.mixins[selector]?["availability"] as? SymbolGraph.Symbol.Availability) {
                     guard !symbolAvailability.availability.isEmpty else { continue }
+                    // For platforms with a fallback option (e.g., Catalyst and iOS), apply the explicit availability annotation of the fallback platform when it is not explicitly available on the primary platform.
+                    DefaultAvailability.fallbackPlatforms.forEach { (fallbackPlatform, inheritedPlatform) in
+                        let fallbackAvailability = symbolAvailability.availability.first(where: {
+                            $0.domain?.rawValue == fallbackPlatform.rawValue
```

Comment:
There is a `matches(_:)` function which we can use to compare a `SymbolGraph.Symbol.Availability.Domain` with `PlatformName` to avoid code duplication (we have this comparison a couple of times in this function): 
https://github.com/apple/swift-docc/blob/e0a862c0ed1c52def62718d6462ac22728047117/Sources/SwiftDocC/Infrastructure/Symbol%20Graph/SymbolGraphLoader.swift#L461-L463

```suggestion
                            $0.matches(fallbackPlatform)
```